### ISSUE TITLE
[git] Selected git repository does not get updated on repository deletion

### ISSUE THREAD

**Issue by kittaakos** [theia-ide/theia#927]
This is a follow up of #899.

Let assume you have repository A, B in your workspace and repository A is selected in the drop-down in the Git tab. Lose the focus from the Git tab (select Files or you name it), delete repository A outside from Theia. Activate the Git tab, repository A is still selected as the active one.

Expected:
B should be selected, A should not be among the repositories.

This glitch does not cause any problem, the git watcher was gracefully disposed. One has to refresh the view manually.

**Comment by kittaakos** [comment]
This could be invalid after #1091. I look into this task.

**Comment by kittaakos** [comment]
I can confirm, the issue still present. The frontend does not receive any notification if the repository gets removed from the workspace. The active repository is still the deleted one. It used to work before #899.

**Comment by kittaakos** [comment]
> It used to work before #899.

No. Way before this... I think it never made it to the `master`. See the removal [here](https://github.com/theia-ide/theia/commit/227a6a1733d1e492b47c738ead48e90bde6ab83b#diff-ac13cf9603943154afffa5c9ce738778).

**Comment by kittaakos** [comment]
@svenefftinge, @akosyakov; shall we treat this behavior as a bug or a missing feature?

I would expect that the UI (frontend state) gets updated after a repository deletion but adding this could be costly performance wise. We could detect if a `.git` folder was deleted from the WS, but that point we cannot check whether it was a repository or not. So we would need to maintain some state on the backend.

**Comment by svenefftinge** [comment]
We can categorize it as a bug. Low priority, though, because it doesn't have bad implications and is a rare corner-case.
